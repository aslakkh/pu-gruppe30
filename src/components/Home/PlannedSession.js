/**
 * Created by jan on 07/03/2017.
 */
import React, { Component } from 'react'
import {Button, ProgressBar, ListGroup, ListGroupItem, Grid,Col,Row} from 'react-bootstrap'
import firebase from 'firebase'


function Test(props){
    return(
        <ListGroup>
            <ListGroupItem>Link 1</ListGroupItem>
            <ListGroupItem onClick={ (link) => props.this.callbackParent(link)}>Link 2</ListGroupItem>
        </ListGroup>


    )
}




export default class PlannedSession extends Component{


    constructor(props){
        super(props)
        this.state={
            desc: "hei",
            emne: this.props.emne
        }

    }
componentWillMount(){
    let that = this
    const userUid = firebase.auth().currentUser.uid;
    firebase.database().ref().child('users/'+userUid+'/courses/'+that.state.emne+'/planned-sessions/').orderByValue().startAt().on('value', snap => {
        console.log(snap.val())
        that.setState({
            sessions: snap.val()
        })
        console.log(snap.val())
})
}

    handleClick(key){
        this.props.callbackParent(key);
    }



    render(){
        return(
            <ListGroup >
                {Object.keys(this.state.sessions).map((key) => {
                    return <ListGroupItem key={key} className="CoursesList">
                        {this.state.sessions[key].goal}
                        {console.log(key)}
                        {new Date(parseInt(key)).toISOString()}
                    </ListGroupItem>
                })}
            </ListGroup>
        )
    }
}