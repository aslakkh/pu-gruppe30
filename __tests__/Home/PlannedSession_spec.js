/**
 * Created by jan on 19/04/2017.
 */
import React from 'react';
import renderer from 'react-test-renderer';
import PlannedSession from '../../src/components/Home/PlannedSession.js';
import TestUtils from "react-addons-test-utils"
import firebase from 'firebase'
import shallow from 'enzyme'
//state.sessions = jest.fn(() => {return {1492786690143: {goal: "grsg"}}});
const courses={
    TES1000:{active: true,
        goals: {dailyGoal:{active: false,
            timeSet: 0,
            timeSpent: 0,
            value: 0},
            monthlyGoal:
                {active:false,
                    timeSet:0,
                    timeSpent:0,
                    value:0},
            weeklyGoal:
                {active:false,
                    timeSet:0,
                    timeSpent:0,
                    value:0}

        }
    }};
const coursesNull=null;
describe("Stopwatch", function () {
    let renderer2;
    let component2;
    renderer2 = TestUtils.createRenderer();
    renderer2.render(
        <PlannedSession emne={"TES1000"} callbackParent={(newState) => this.onChildChanged(newState) }/>
    );
    component2 = renderer2.getRenderOutput();
    it("One course", () => {
        expect(component2).toMatchSnapshot();
    });
    it('Open set old session', () => {
        const wrapper = shallow(<AddSession courseID={"TES1000"}/>);
        wrapper.find('Button').first().simulate('click');
        wrapper.find('Button').at(1).simulate('click');



    })
});