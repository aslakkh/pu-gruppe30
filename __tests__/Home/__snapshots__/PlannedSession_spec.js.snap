// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Stopwatch One course 1`] = `
<div>
  <h4>
    Planned Sessions
  </h4>
  <ListGroup
    bsClass="list-group"
  >
    <ListGroupItem
      bsClass="list-group-item"
      className=""
      listItem={false}
      onClick={[Function]}
    >
      <div
        style={
          Object {
            "fontSize": "130%",
          }
        }
      >
        hello
      </div>
      <div
        style={
          Object {
            "color": "#999999",
          }
        }
      >
        Wed Apr 26 2017
      </div>
    </ListGroupItem>
    <ListGroupItem
      bsClass="list-group-item"
      className=""
      listItem={false}
      onClick={[Function]}
    >
      <div
        style={
          Object {
            "fontSize": "130%",
          }
        }
      >
        hello
      </div>
      <div
        style={
          Object {
            "color": "#999999",
          }
        }
      >
        Wed Apr 26 2017
      </div>
    </ListGroupItem>
  </ListGroup>
</div>
`;

exports[`Stopwatch Welcome renders hello world 1`] = `
<h4>
  No planned sessions
</h4>
`;

exports[`Stopwatch new course 1`] = `
<div>
  <h4>
    Planned Sessions
  </h4>
  <ListGroup
    bsClass="list-group"
  >
    <ListGroupItem
      bsClass="list-group-item"
      className=""
      listItem={false}
      onClick={[Function]}
    >
      <div
        style={
          Object {
            "fontSize": "130%",
          }
        }
      >
        hello
      </div>
      <div
        style={
          Object {
            "color": "#999999",
          }
        }
      >
        Wed Apr 26 2017
      </div>
    </ListGroupItem>
    <ListGroupItem
      bsClass="list-group-item"
      className=""
      listItem={false}
      onClick={[Function]}
    >
      <div
        style={
          Object {
            "fontSize": "130%",
          }
        }
      >
        hello
      </div>
      <div
        style={
          Object {
            "color": "#999999",
          }
        }
      >
        Wed Apr 26 2017
      </div>
    </ListGroupItem>
  </ListGroup>
</div>
`;
